lib/src/lints/flutter/proper_controller_dispose.dart

```diff
@@ -31,91 +32,166 @@ class ProperControllerDispose extends DartLintRule {
-              }
-            }
-            return null;
+            return switch (e.expression) {
+              final MethodInvocation exp when exp.target is SimpleIdentifier =>
+                (exp.target! as SimpleIdentifier).name,
+              _ => null
+            };
           })
-          .whereType<String>()
+          .whereNotNull()
           .toList();
 
-      for (final controllerDeclaration in controllerDeclarations) {
-        if (!disposeStatementTargetNames
-            .contains(controllerDeclaration.name.lexeme)) {
-          reporter.reportErrorForElement(
-            code,
-            controllerDeclaration.declaredElement!,
-            [controllerDeclaration.name.lexeme],
-          );
+      for (final variable in controllerVariables) {
+        if (!disposeStatementTargetNames.contains(variable.name)) {
+          for (final variable in controllerVariables) {
+            final variableName = variable.name;
+            reporter.reportErrorForElement(
+              code,
+              variable,
+              [variableName],
+            );
+          }
+        }
+      }
+    });
+  }
+
+  @override
+  List<Fix> getFixes() => [_ProperControllerDisposeFix()];
+}
+
+class _ProperControllerDisposeFix extends DartFix {
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ChangeReporter reporter,
+    CustomLintContext context,
+    AnalysisError analysisError,
+    List<AnalysisError> others,
+  ) {
+    context.registry.addFieldDeclaration((node) {
+      if (!analysisError.sourceRange.intersects(node.sourceRange)) return;
+
+      final parent = node.parent! as ClassDeclaration;
+
+      final disposeFunctionBody =
+          parent.members.findMethodDeclarationByName('dispose')?.body;
+
+      final changeBuilder = reporter.createChangeBuilder(
+        message: 'Dispose controller',
+        priority: 80,
+      );
+
+      final controllerVariables =
+          node.fields.variables.map((e) => e.declaredElement!).where(
+                (e) =>
+                    changeNotifierChecker.isSuperTypeOf(e.type) ||
+                    disposableControllerChecker.isExactlyType(e.type),
+              );
+
+      if (disposeFunctionBody == null) {
+        final initStateMethod = parent.members.findMethodDeclarationByName(
+          'initState',
+        );
+
+        final buildMethod = parent.members.findMethodDeclarationByName('build');
+
+        final (int offset, bool addNewLineAtTheStart, bool addNewLineAtTheEnd) =
+            switch ((initStateMethod, buildMethod)) {
+          (null, null) => (node.end, true, false),
+          (null, final buildMethod?) => (buildMethod.offset, true, true),
+          (final initStateMethod?, null) => (initStateMethod.end, true, false),
+          (final _?, final buildMethod?) => (buildMethod.offset, false, true),
+        };
+
+        /// Always pick the first variable.
+        final variableName = controllerVariables.first.name;
```

Comment:
Hmm.. not sure why we always use the first one

Reply:
I think the default analyzer's add 'require' param does the same sometimes, that's why I use the first one. We can check for the intersection point for the variable being the assist triggered upon. If you have any other ideas please comment.